### PR TITLE
feat(dark-mode): refactor branded colors utility

### PR DESCRIPTION
**Pull request by knajjars**
```ts
const {
  theme,
  text,
  bg,
  fill,
  border,
  contrastText,
  contrastBg,
  contrastFill,
  contrastBorder,
} = useBrandedColors(agentId);
```

Refactors the logic for using branded colors for assistants. This takes a more scalable approach that it is easy to scale and use across the app, it is also much more efficient to run.

Chat _ Cohere · 5 44pm · 08-01

### CODE REVIEW

File: src/interfaces/assistants_web/src/components/Shared/Button/Button.stories.tsx [hunk 1]
```diff
@@ -73,6 +88,11 @@ export const AllStyles: StoryFn<typeof Button> = () => (
         <Button kind="outline" icon="add" label="Outline" theme="coral" />
         <Button kind="outline" icon="add" label="Outline" theme="quartz" />
         <Button kind="outline" icon="add" label="Outline" theme="mushroom" />
+        <Button kind="outline" icon="add" label="Outline" theme="danger" />
+        <Button kind="outline" icon="add" label="Outline" theme="evolved-blue" />
+        <Button kind="outline" icon="add" label="Outline" theme="evolved-mushroom" />
+        <Button kind="outline" icon="add" label="Outline" theme="evolved-quartz" />
+        <Button kind="outline" icon="add" label="Outline" theme="green" />
```
Comment (tomtobac) [inline src/interfaces/assistants_web/src/components/Shared/Button/Button.stories.tsx:95]: We dont have this themes...

image

File: src/interfaces/assistants_web/src/utils/cohereColors.ts [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+  options: {
+    text?: boolean;
+    fill?: boolean;
+    border?: boolean;
+    background?: boolean;
+    contrastText?: boolean;
+  },
+  variant?: { hover?: boolean }
+): string => {
+  if (id === undefined) {
```
Comment (tomtobac) [inline src/interfaces/assistants_web/src/utils/cohereColors.ts:190]: so many deep ifs.. 😱

File: src/interfaces/assistants_web/src/components/Shared/Button/Button.stories.tsx (hunk 1 above)
Comment (knajjars) [reply src/interfaces/assistants_web/src/components/Shared/Button/Button.stories.tsx:95]: we need them for assistants custom colors

File: src/interfaces/assistants_web/src/utils/cohereColors.ts (hunk 2 above)
Comment (knajjars) [reply src/interfaces/assistants_web/src/utils/cohereColors.ts:190]: yes, I have an idea to refactor all of this. On follow up PR, let's see

**Review by tomtobac — APPROVED** [review]

File: src/interfaces/assistants_web/src/components/Shared/Button/Button.stories.tsx (hunk 1 above)
Comment (tomtobac) [reply src/interfaces/assistants_web/src/components/Shared/Button/Button.stories.tsx:95]: I think we should called them `assistant-accent-{color}`

Comment (tomtobac) [reply src/interfaces/assistants_web/src/components/Shared/Button/Button.stories.tsx:95]: otherwise seems like its part of the theme

Comment (abimacarmes) [reply src/interfaces/assistants_web/src/components/Shared/Button/Button.stories.tsx:95]: can we ask them to conform assistants custom colors to the existing themes?

**Review by abimacarmes — APPROVED** [review]

File: src/interfaces/assistants_web/src/components/Shared/Button/Button.stories.tsx (hunk 1 above)
Comment (knajjars) [reply src/interfaces/assistants_web/src/components/Shared/Button/Button.stories.tsx:95]: lets keep and remove if needed later, but it is more consistent to have all themes for all components

**Review by tomtobac — APPROVED** [review]

**Pull request merged**